linera-shared-storage/proto/key_value_store.proto

```diff
@@ -0,0 +1,67 @@
+syntax = "proto3";
+package key_value_store.v1;
+
+message KeyValue {
+  bytes key = 1;
+  bytes value = 2;
+}
+
+message Statement {
+  oneof Operation {
+    bytes delete = 1;
+    KeyValue put = 2;
+    bytes delete_prefix = 3;
+  }
+}
+
+message BatchBaseKey {
+  repeated Statement statements = 1;
+  bytes base_key = 2;
+}
+
+message UnitType {
+}
+
+message StoreRequest {
+  oneof query {
+    bytes read_value = 1;
+    bytes contains_key = 2;
+    Keys read_multi_values = 3;
+    bytes find_keys_by_prefix = 4;
+    bytes find_key_values_by_prefix = 5;
+    BatchBaseKey write_batch = 6;
+    bytes clear_journal = 7;
+  }
+}
+
+message OptValue {
+  optional bytes value = 1;
+}
+
+message OptValues {
+  repeated OptValue values = 1;
+}
+
+message Keys {
+  repeated bytes keys = 1;
+}
+
+message KeyValues {
+  repeated KeyValue key_values = 1;
+}
+
+message StoreReply {
+  oneof reply {
+    OptValue read_value = 1;
+    bool contains_key = 2;
+    OptValues read_multi_values = 3;
+    Keys find_keys_by_prefix = 4;
+    KeyValues find_key_values_by_prefix = 5;
+    UnitType write_batch = 6;
+    UnitType clear_journal = 7;
+  }
+}
+
+service StoreProcessor {
+  rpc StoreProcess (StoreRequest) returns (StoreReply) {}
```

Comment:
Why not create one RPC entrypoint per method instead of doing the RPC dispatch by hand in `StoreRequest` and `StoreReply` ?

Reply:
Sorry, missed that one. I think we can do that, though I am not completely sure. My question is why?

Reply:
That's how GRPC is supposed to be used

Reply:
(Notably this allows matching the response types with their respective query types.)

Reply:
You were right, the code is now much simpler.